src/Symfony/Bundle/FrameworkBundle/CacheWarmer/AnnotationsCacheWarmer.php

```diff
@@ -41,71 +36,42 @@ class AnnotationsCacheWarmer implements CacheWarmerInterface
      */
     public function __construct(Reader $annotationReader, $phpArrayFile, CacheItemPoolInterface $fallbackPool)
     {
+        parent::__construct($phpArrayFile, $fallbackPool);
         $this->annotationReader = $annotationReader;
-        $this->phpArrayFile = $phpArrayFile;
-        if (!$fallbackPool instanceof AdapterInterface) {
-            $fallbackPool = new ProxyAdapter($fallbackPool);
-        }
-        $this->fallbackPool = $fallbackPool;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function warmUp($cacheDir)
+    protected function doWarmUp($cacheDir, PhpArrayAdapter $phpArrayAdapter, ArrayAdapter $arrayAdapter)
     {
-        $adapter = new PhpArrayAdapter($this->phpArrayFile, $this->fallbackPool);
         $annotatedClassPatterns = $cacheDir.'/annotations.map';
 
         if (!is_file($annotatedClassPatterns)) {
-            $adapter->warmUp(array());
+            $phpArrayAdapter->warmUp(array());
 
-            return;
+            return false;
```

Comment:
this makes me think that doWarmUp doesn't need the $phpArrayAdapter argument: it's used only here, but returning !== false here would ship the same behavior without the need for this arg, isn't it?
which then triggers: shouldn't we have a clean "bool" return type for all dowarmup methods? ie return true on success? (and here also then)